src/lib/server/trunk.c

```diff
@@ -3167,7 +3167,7 @@ static void trunk_connection_enter_active(fr_trunk_connection_t *tconn)
 	/*
 	 *	Rebalance requests
 	 */
-	trunk_rebalance(trunk);
+	if (!trunk->conf.block_rebalance) trunk_rebalance(trunk);
```

Comment:
It's not enough.. if the connection closed then the request would still get moved to another connection.  It's probably be easier to remove the rebalance and to rework the code so that balancing was done in fr_trunk_connection_pop_request instead.  The difficulty is how to actually do that balancing.  I guess we could just allow as many requests to be popped as the max active request limit for the connection? I think it's 255 for RADIUS because we reserve one... one request per connection for "WAIT"ing redis... and I suppose as many requests it takes to fill the write buffer for LDAP.  There the muxer would be self-limiting and stop popping requests once the write buffer was fille.

Reply:
Further testing has shown that `per_connection_max` is doing what is needed, so this change is not required.  Not sure why it didn't work first time through - probably something to do with the timing of marking the trunk requests as complete - that did change a number of times to get all the relevant structure lifetimes correct.